### PR TITLE
Push notification handler

### PR DESCRIPTION
**Pull request by Krakaw**
### References Issues:
https://app.asana.com/0/1149047845558135/1146369744544823
### Description:
When a push notification is selected it pings the server to increment the open count
### Environment Variables:
 * No change

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -54,13 +55,48 @@ export default class App extends Component {
         name: 'Event messages',
         sound: true,
         vibrate: true,
+      }).then(r => {
+
```
Comment (coka) [inline App.js]: What is this used for?

File: App.js [hunk 2]
```diff
@@ -54,13 +55,48 @@ export default class App extends Component {
...
+    "broadcast_id": "asb",
+    "event_id": "eeeeee",
+  },
+  "isMultiple": false,
+  "notificationId": -1253711886,
+  "origin": "selected",
+  "remote": true,
+  "userText": null,
+}
+     */
```
Comment (coka) [inline App.js]: We should avoid committing commented code.

File: App.js [hunk 3]
```diff
@@ -54,13 +55,48 @@ export default class App extends Component {
...
+    if (id && origin === 'selected') {
+      //They opened it
```
Comment (coka) [inline App.js]: Maybe there can be a variable above the conditional, like:

```javascript
const notificationOpened = id && origin === 'selected')
```

File: App.js [hunk 4]
```diff
@@ -54,13 +55,48 @@ export default class App extends Component {
...
+      server
+        .broadcasts
+        .opened({ id })
+        .then(_r => {
```
Comment (coka) [inline App.js]: I honestly don't think this empty `then()` does anything. We can have a `catch()` without a `then()`.

File: App.js [hunk 5]
```diff
@@ -54,13 +55,48 @@ export default class App extends Component {
...
+        }).catch(e => {
+        console.error(e)
       })
     }
+    this.setState({ notification: notification })
```
Comment (coka) [inline App.js]: This state isn't a factor anywhere else. Could be a leftover from debugging?

File: App.js (hunk 1 above)
Comment (Krakaw) [reply App.js]: Purely to It can be ignored.

File: App.js (hunk 3 above)
Comment (Krakaw) [reply App.js]: It is defined above, I'm not sure I follow though?
```
    const { broadcast_id: id = null } = data
```

File: App.js (hunk 4 above)
Comment (Krakaw) [reply App.js]: You're correct, removed.

File: App.js (hunk 5 above)
Comment (Krakaw) [reply App.js]: Correct.

File: App.js (hunk 2 above)
Comment (Krakaw) [reply App.js]: Agreed.

**Comment by Krakaw** [comment]
> Not that I personally mind, but the `config.js` change would be a good opportunity to provide an example of what a PR description with environment variable changes should look like.

Since there weren't any new ENV's added, I didn't update the environment variable section.

File: App.js (hunk 3 above)
Comment (coka) [reply App.js]: The comment `//They opened it` clarifies the conditional logic in the line above. What I'm suggesting is to give a descriptive name to the result of `id && origin === ...`, removing the need for a comment.

Comment (Krakaw) [reply App.js]: Pushed a more descriptive variable.

**Pull request merged**